### ISSUE TITLE
Provider icons in top right not working.

### ISSUE THREAD

**Issue by OzGav** [music-assistant/hass-music-assistant#521]
### What version of Music Assistant has the issue?

2022.6.6

### The problem

I understand the icons in the top right (in my case local files system and spotify) should filter the song listing below. When I hover over it the cursor changes to the finger to suggest it is clickable. The tooltip with the source name appears. However when I click on those icons nothing appears to happen?

### How to reproduce

As above

### Relevant log output

Not sure if this is relevant?

```
Logger: music_assistant.music
Source: /usr/local/lib/python3.9/site-packages/music_assistant/controllers/music/__init__.py:121
First occurred: 4:11:45 PM (1 occurrences)
Last logged: 4:11:45 PM

Provider ProviderType.DATABASE is not available
```

### What version of Home Assistant Core are your running

2022.6.6

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Raspberry Pi

**Comment by marcelveldt** [comment]
Please explain with some screenshots what you mean because your report is not very clear yet to me. If you refer to the provider icons (e.g. Spotify) in the top right in the "info" view. Those are not meant for filtering but should bring you to the webinterface (or app) of that provider. Looks like that got broken somehow so that is a bug.

You are talking about filtering however and I have no idea what you want filtered here.

**Comment by OzGav** [comment]
This is the Album view but same issue if I go though the Artist view. In the top right there are the two icons below the vertical ellipsis. What are they supposed to do if anything?

My core issue is I have this album (amongst many) which I have on my NAS in FLAC. I see that MA has also pulled down the songs from Spotify, The unfortunate problem in this situation is that Spotify has called the Album the same name but has appended (Remaster) to all the song titles so everything is appearing duplicated. I would like to hide the spotify entries.

![image](https://user-images.githubusercontent.com/19848947/175878731-2b21557c-9f5e-40a4-a4e8-4eff2f17c6fc.png)

**Comment by OzGav** [comment]
Yes they are the ones. Not 100% sure what they are for?

**Comment by marcelveldt** [comment]
Those icons should take you to the web interface (or app) of the provider for the item (album in this case) but this is not working (which is a bug).

Can you please check this album on Spotify if this is whole album is considered a remaster ?
In that case there's a bug. If not this is not a bug.

A remaster is not the same version as a non-remaster so the matcher logic doing exactly what it should do.
It simply may not match an original to a remaster unless explicitly told so.

What I find strange is that original (non remaster) tracks are in the same albumlisting now hence my question if you can have a quick look at Spotify if the album is a remaster or not.

Looking at the duration of the tracks these are in fact different versions.

**Comment by OzGav** [comment]
Its probably a Spotify database issue. On Spotify the Alubum name is Abacab. The cover art shown is the original album but the tracks all have either "2007 Digital Remaster" or "Remaster 2007" appended. This remaster was done as part of a boxset https://en.wikipedia.org/wiki/Genesis_1976%E2%80%931982 so it should have different IDs https://musicbrainz.org/release-group/4488cc99-a9a1-3f10-adcc-043a18c53078

**Comment by OzGav** [comment]
Also I am only new to MA but I see there is a VERSIONS tab which shows 2. Which I go into that view clicking on the two doesn't seem to do anything? Maybe one option to help with this would be to SORT BY VERSION (and then TRACK NUMBER)?

**Comment by marcelveldt** [comment]
Exactly! So MA is doing exactly what it should do and it doesn't match the remaster with the non-remaster.
Strange that Spotify has catalogued the album wrong then.

I'm thinking out loud how to prevent such cases. The only thing I can think of:

1) Ability to manually assign the tracks to a different album (or edit the spotify matched album)
2) Detect if all tracks of an album have a different version and in that case assume the album is a different version.

While 2 may be nice from user perspective the actual error is hidden so maybe a bit dangerous.

**Comment by marcelveldt** [comment]
> Also I am only new to MA but I see there is a VERSIONS tab which shows 2. Which I go into that view clicking on the two different albums doesn't seem to do anything? Maybe one option to help with this would be to SORT BY VERSION (and then TRACK NUMBER)?

Versions tab should show any other different versions of this album (such as Remaster, Live etc.) on all providers.
Sometimes this doesn't make a lot of sensor if it just shows the versions that are already matched and it could in fact be hidden

**Comment by marcelveldt** [comment]
> Maybe one option to help with this would be to SORT BY VERSION (and then TRACK NUMBER)?

That does sound viable as a workaround but it still itches that the tracks of another album version now are in the album listing of another album version... Guess we can't fix Spotify's metadata so maybe we should indeed work around it ?

**Comment by erkr** [comment]
Hi, music assistant blends songs from various sources into one jukebox. It automatically slices the best quality available. To my understanding these icons represent the sources used. In your case songs from both Spotify and local media.

**Comment by marcelveldt** [comment]
> Hi, music assistant blends songs from various sources into one jukebox. It automatically slices the best quality available. To my understanding these icons represent the sources used. In your case songs from both Spotify and local media.

Erik, see the above conversation. The icons not working is just a very small bug. There is another issue going on that tracks of a remastered album version are listed together with tracks of the not-remastered (original) album.

**Comment by marcelveldt** [comment]
> Thanks @marcelveldt . Maybe you could have it so that if you click on the album in the VERSIONS tab then you see the list of tracks from that ALBUM? 

Great idea! That is something easy to implement. 

> I just double checked my MB ID's (to make sure I had not made the error!) and I am using a release from 1988 (Recording ID dbc137d6-ac33-4503-a9d9-df63e7f841e3 and Release ID f7eb677a-cf39-47d7-b644-37c3cde348ce). Does Spotify have the MB ID's in their metadata? If so can you see what release they are using and is MA using RELEASE ID to match? I ask because if using RECORDING ID then I believe they will match because that number will be the same....

Spotify uses the UPC and that is what MA is using to match... So I think Spotify has the wrong UPC set on the album ?

**Comment by OzGav** [comment]
> Great idea! That is something easy to implement.

Hang on I think you have already done this!! I just went to another album "1" by The Beatles. I see there are TWO versions. Now I can see when I click on those that the source icon in the top right changes (that is I only see ONE of them) and when I go back to the TRACKS tab I only see the tracks for the selected album. In this case when I select the SPOTIFY version it is also a remastered release but MA knows it is a different version. So in this case I am only seeing the tracks related to the version I select. 

However, in the Genesis example I see two versions but when I click on EITHER of them I still see BOTH source icons. So if MA was matching the same album and then combing all the tracks why is it showing TWO versions? Maybe a good reason I am just talking out loud to you :-) 

While I am throwing feature requests at you; I notice I can't tell in the VERSION tab which one I have selected so I don't know beforehand which is the selected version. Could you add an icon like in this image

![image](https://user-images.githubusercontent.com/19848947/175927052-fa025353-f046-4388-aafd-5a77e224ee2d.png)

**Comment by OzGav** [comment]
> Spotify uses the UPC and that is what MA is using to match

Ahhh... I see in MB that there is no barcode/UPC for the release I am using...

**Comment by erkr** [comment]
I adapted the title to reflect the small bug that the provider icons in top right don't work.

Not sure from the discussion if the doubles (original& remastered) is a Spotify mistake or also a MA bug? 

@marcelveldt  Please decide on creating an extra issue or spawn a feature request for the test of the discussion

**Comment by marcelveldt** [comment]
> Hang on I think you have already done this!! I just went to another album "1" by The Beatles. I see there are TWO versions. Now I can see when I click on those that the source icon in the top right changes (that is I only see ONE of them) and when I go back to the TRACKS tab I only see the tracks for the selected album. In this case when I select the SPOTIFY version it is also a remastered release but MA knows it is a different version. So in this case I am only seeing the tracks related to the version I select.

Correct, it already works this way but it still auto matches the same album. So if the album 1 on provider X is matched to album 2 on provider Y it will still show the combined (database) view. That's something I can adjust so that if you click the provider-specific album from the versions tab, it may only view that exact provider version/copy.

> However, in the Genesis example I see two versions but when I click on EITHER of them I still see BOTH source icons. So if MA was matching the same album and then combing all the tracks why is it showing TWO versions? Maybe a good reason I am just talking out loud to you :-)

Yes, because of what I just told you. The word Version is a bit misleading in this case. It can mean a "provider copy" (the same album on provider X, Y , Z) and/or a "album version" (such as Live, Remaster etc.).
The version tab just queries all providers for the album name and lists them, regardless of provider, version. 


> While I am throwing feature requests at you; I notice I can't tell in the VERSION tab which one I have selected so I don't know beforehand which is the selected version. Could you add an icon like in this image

Set the view to "list" and that is what you (should) get...


![Schermafbeelding 2022-06-27 om 14 07 19](https://user-images.githubusercontent.com/6389780/175937469-258500d2-74df-4e30-b91a-2e4e28e0a197.png)

**Comment by marcelveldt** [comment]
BTW: I just discovered that this issue exists globally on Spotify. They just consider the album the same, regardless if the tracks are remastered or not. 

So the solution will be to either sort the track version separate from the track number or detect all tracks are remastered and consider the album to be the remaster.

**Comment by OzGav** [comment]
Thanks the list view is better for me. Yes I am seeing a lot of albums where Spotify has the tracks listed as "remastered". 

Also, just to make things (maybe) more confusing for you. I have found an example where the album name is the same and all the tracks except one have remaster in the name. The one one without remaster happens to be the only track I have from that album but it is not matching. So same album, title, disc number, track number and duration but no match. So I don't know what is going on there?

**Comment by marcelveldt** [comment]
Can you give me the spotify id from that album and track ? You can find these in the Spotify app/webplayer in the "share link" logic. Or wait for me to fix the bug so that the icon links work again ;-)

And a screenshot would be nice of the offending album listing.

**Comment by OzGav** [comment]
Yes. Spotify link https://open.spotify.com/album/31nRnHI1uSDZ0POYPPJcUH?si=7fPKwHviRYik6qpuu9wqXQ It is track number 6 as per this screen shot

![image](https://user-images.githubusercontent.com/19848947/175940894-d8002ab8-f749-42fa-9603-5bed2cfd982c.png)

**Comment by marcelveldt** [comment]
That is a strange one indeed. It should have matched this one. Now I'm curious if the isrc of your track on disk is the same (or not) as the Spotify one. This may as well be Spotify's auto track replacement at work. Maybe it just considers a remaster not as a different version and delivers the same isrc for both versions ? That also explains the high number of streams for that track (13 mln)

**Comment by OzGav** [comment]
The ISRC in my file is GBAZP9900012

**Comment by marcelveldt** [comment]
Ok, I looked at the raw data from Spotify and Spotify has the data correct so the issue must be somewhere within MA's matching logic. I think it's completely ignoring the album version somehow. 

So... long story short: we have a bug to solve.
Thanks for the help diagnosing!

I'll let you know once I have something to test.

**Comment by marcelveldt** [comment]
I fixed the issue with the provider icons in 2022.6.7. The other issue needs some more testing with 2022.6.7, don't think I resolved that one completely now.

**Comment by OzGav** [comment]
Yes thanks @marcelveldt those other issues are resolved. However, this example of The Go-Betweens is still not matching. Please let me know if I can do anything more. I have another (what I think is a different matching) issue that i can raise as well if you would like.

**Comment by marcelveldt** [comment]
fixed in 2022.6.10 - please reopen if not resolved, thanks!
